nupic/research/spatial_pooler.py

```diff
@@ -246,31 +298,30 @@ def __init__(self,
     self._iterationNum = 0
     self._iterationLearnNum = 0
 
-    # Store the set of all inputs that are within each column's potential pool.
-    # 'potentialPools' is a matrix, whose rows represent cortical columns, and
-    # whose columns represent the input bits. if potentialPools[i][j] == 1,
+    # Store the set of all inputs within each columns potential pool as a
+    # single adjacency matrix such that matrix rows map to cortical columns,
+    # and matrix columns map to input buts.  If potentialPools[i][j] == 1,
     # then input bit 'j' is in column 'i's potential pool. A column can only be
-    # connected to inputs in its potential pool. The indices refer to a
-    # flattened version of both the inputs and columns. Namely, irrespective
-    # of the topology of the inputs and columns, they are treated as being a
-    # one dimensional array. Since a column is typically connected to only a
-    # subset of the inputs, many of the entries in the matrix are 0. Therefore
-    # the potentialPool matrix is stored using the SparseBinaryMatrix
-    # class, to reduce memory footprint and computation time of algorithms that
-    # require iterating over the data structure.
-    self._potentialPools = SparseBinaryMatrix(numInputs)
+    # connected to inputs in its potential pool.  Here, BinaryCorticalColumns
+    # is used to provide cortical column-centric semantics for what is
+    # otherwise a sparse binary matrix implementation.  Sparse binary matrix is
+    # used as an optimization since a column will only be connected to a small
+    # fraction of input bits.
+    self._potentialPools = BinaryCorticalColumns(numInputs)
     self._potentialPools.resize(numColumns, numInputs)
```

Comment:
can we already initialize this with numColumns, numInputs? as here https://github.com/numenta/nupic/pull/2550/files#diff-323276e3bd86539a560868bd2961f1d7R324 ?